pkg/gke/gke.go

```diff
@@ -0,0 +1,127 @@
+package gke
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func BuildCrbObject() (*unstructured.Unstructured, error) {
+	user, err := getActiveUser()
+	if err != nil {
+		return nil, err
+	}
+
+	crb := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "ClusterRoleBinding",
+			"apiVersion": "rbac.authorization.k8s.io/v1beta1",
+			"metadata": map[string]interface{}{
+				"name": "kubeapps-cluster-admin",
+			},
+			"roleRef": map[string]interface{}{
+				"apiGroup": "rbac.authorization.k8s.io",
+				"kind":     "ClusterRole",
+				"name":     "cluster-admin",
+			},
+			"subjects": []map[string]interface{}{
+				{
+					"apiGroup": "rbac.authorization.k8s.io",
+					"kind":     "User",
+					"name":     user,
+				},
+			},
+		},
+	}
+	return crb, nil
+}
+
+func getActiveUser() (string, error) {
+	gcloudPath, err := sdkConfigPath()
+	if err != nil {
+		return "", err
+	}
+	activeConfigPath := filepath.Join(gcloudPath, "active_config")
+	activeConfig, err := ioutil.ReadFile(activeConfigPath)
+	if err != nil {
+		return "", err
+	}
+	configPath := filepath.Join(gcloudPath, "configurations")
+	f, err := os.Open(filepath.Join(configPath, "config_"+string(activeConfig)))
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	ini, err := parseINI(f)
```

Comment:
maybe we don't need to reinvent the wheel:
https://github.com/go-ini/ini

Reply:
Ah cool. I haven't known it

Reply:
imported that lib. Thanks @andresmgot